### PR TITLE
Add Newsletter Signup

### LINKED ISSUE #7450: Store Profiler - Store details step: include newsletter sign up
**Description:** if new users want to sign up for our email marketing communications, they need to do it through an inbox notification. This can easily go unnoticed and the notification redirects people to the Mailchimp confirmation page, which is a terrible experience. Signing up for email marketing should be a seamless experience (you just provide your email and that’s it!). We should try to incorporate the sign-up in the store setup experience and capture the email address in a way that would allow us to reuse it in other parts of the setup flow. For example, to streamline the WCPay account creation process. 

**Acceptance criteria:**
- [ ] Remove the inbox notification `WC_Admin_Notes_Onboarding_Email_Marketing.php`
- [ ] Move the `I'm setting up a store for a client` checkbox to the Business Details step

- [ ] Introduce the email address field in the Store Details step. Can we capture the email address in a way that would allow us to reuse it in other parts of the setup flow?
- [ ] Introduce a new checkbox below the email address (copy TBD). It should be checked by default

- [ ] If the user clicks "Continue" with the checkbox checked, use the Mailchimp API to create a new subscriber.

**Event tracking:**
- Add a new event prop called `email_signup` to `wcadmin_storeprofiler_store_details_continue`. This prop should record if the user moved forward with the email marketing sign up: True|False
- Move the `setup_client` prop to `wcadmin_storeprofiler_store_business_details_continue_variant`. To follow the new placement of the `I'm setting up a store for a client` checkbox: True|False

### PR DESCRIPTION
**Pull request by adrianduffell**
Fixes #7450

This is a feature branch including PRs #7552, #7579, #7570, #7590, #7595.

It adds newsletter signup functionality to the first step of the profiler. This necessitates moving the "client" checkbox to the Business Details step.

A daily background process is added via the daily cron that processes the email address and marketing consent checkbox. When consented, the email address is added to the WooCommerce mailing list using an endpoint hosted at woocommere.com. There is some resiliency where if the endpoint is unavailable (e.g. due to network error), it is retried the following day.

### Screenshots

https://user-images.githubusercontent.com/9312929/131343078-d1975b0a-f6af-4632-9b47-4d0f7882bc48.mov

### Detailed test instructions:

- Start OBW and set up your browser console to monitor tracks. To do this, run `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
- Observe "Get tips, product updates and inspiration straight to your mailbox" checkbox and "Email address" field in the Store Details step.
- Checking the checkbox should make the email field required, you should not be able to continue if it's not filled.
- Fill in the email address field with a valid email and click on continue.
- Observe in the track `wcadmin_storeprofiler_store_details_continue` with prop `email_signup` that appropriately flags if the user agreed to receive marketing emails.
- Continue until Business Features step.
- Observe the "I'm setting up a store for a client" checkbox in the step.
- Click on continue.
- Observe in the track `wcadmin_storeprofiler_store_business_details_continue_variant` with prop `setup_client` that appropriately flags if the user is setting up store for a client.

No changelog for this PR (multiple changelings already added in feature branch).

### CODE REVIEW

File: TESTING-INSTRUCTIONS.md [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 ## Unreleased
 
+### Add Newsletter Signup #7601
+
+- Start OBW and set up your browser console to monitor tracks. To do this, run `localStorage.setItem( 'debug', 'wc-admin:tracks' );`
```
Comment (moon0326) [inline TESTING-INSTRUCTIONS.md:7]: I think it should be `localStorage.setItem( 'debug', 'wc-admin:*' )` I wasn't able to get the debug messages show up with ``localStorage.setItem( 'debug', 'wc-admin:tracks' );``

**Review by moon0326 — APPROVED** [review]
Thank you @adrianduffell for putting this PR 👍 Everything LGTM and tested well! 🎉 🚀 

I left a comment for the debug command. 

Pre-approving ✅

File: TESTING-INSTRUCTIONS.md (hunk 1 above)
Comment (adrianduffell) [reply TESTING-INSTRUCTIONS.md]: Good catch! ⚾

I updated it in 8c61c00

Comment (adrianduffell) [reply TESTING-INSTRUCTIONS.md]: And fixed a cop- paste error with that in 7114dc3 😅

**Pull request merged**